ex-4-quicksort/quicksort.js

```diff
@@ -0,0 +1,30 @@
+function quickSort( arr ) {
+
+    if ( arr.length < 2 ) {
+        return arr;
+    }
+
+    const pivot = arr[arr.length - 1];
+    let leftArr = [];
+    let rightArr = [];
+
+    for (let i = 0; i < arr.length - 1; i++) {
+        if ( arr[i] < pivot ) {
+            leftArr.push(arr[i]);
+        } else {
+            rightArr.push(arr[i]);
+        }
+    }
+
+    if ( leftArr.length > 0 && rightArr.length > 0 ) {
+        return [...quickSort( leftArr ), pivot, ...quickSort( rightArr )];
+    } else if ( leftArr.length > 0 ) {
+        return [...quickSort(leftArr), pivot];
+    } else {
+        return [pivot, ...quickSort( rightArr )];
+    }
```

Comment:
🤯

Reply:
>🤯

😆 Thanks for the push.  I've cleaned it up a bit: https://github.com/mdmoore/grokking-algorithms/pull/6/commits/8fe9920550251ad7563513d3f7617c68a3e7de1f